hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1667,6 +1670,48 @@ private void putAllInternal(@Nonnull Map<? extends K, ? extends V> map,
         }
     }
 
+    public CompletableFuture<Void> putAllWithMetadataAsync(@Nonnull Collection<? extends EntryView<K, V>> entries) {
```

Comment:
Can you please make this method in sync with the `putAllAsync`. If there is enhancement, please submit to both. I see for example the following differences:

1. putAllAsync has null checks for key and values (see [line](https://github.com/hazelcast/hazelcast/blob/v5.3.1/hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java#L1624))
2. putAllAsync also updates the nearcache (see [this](https://github.com/hazelcast/hazelcast/blob/v5.3.1/hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java#L1645
3. ) line )

Reply:
Added 574a212de4a87b9768108a23ce9587c5717e66ce
However I am not sure how to test it - I don't see any client test for put/putAll with combination of near cache.

Reply:
E.g. test case: enable near cache, populate map which will fill the near cache, and then do putAll, and observe that the near cache stats invalidations is increased.